File: refactortalentscrm_ui/src/services/NoteService.js
import http from "../http-common";

class NoteService {
  getAllNotes(searchDTO) {
    console.log(searchDTO)
    return this.getRequest(`/note/notes`, searchDTO);
  }

  get(noteId) {
    return this.getRequest(`/note/${noteId}`, null);
  }

  findByField(matchData) {
    return this.getRequest(`/note?field=${matchData}`, null);
  }

  addNote(data) {
    return http.post("/note/addNote", data);
  }

  update(data) {
  	return http.post("/note/updateNote", data);
  }
  
  uploadImage(data,noteId) {
  	return http.postForm("/note/uploadImage/"+noteId, data);
  }




	postRequest(url, data) {
		return http.post(url, data);
      };

	getRequest(url, params) {
        return http.get(url, {
        	params: params
        });
    };

}

export default new NoteService();

File: src/main/java/com/refactortalentscrm/service/NoteService.java
package com.refactortalentscrm.service;

import java.util.List;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.domain.Specification;
import org.springframework.http.ResponseEntity;

import com.refactortalentscrm.domain.Note;
import com.refactortalentscrm.dto.NoteDTO;
import com.refactortalentscrm.dto.NoteSearchDTO;
import com.refactortalentscrm.dto.NotePageDTO;
import com.refactortalentscrm.dto.NoteConvertCriteriaDTO;
import com.refactortalentscrm.service.GenericService;
import com.refactortalentscrm.dto.common.RequestDTO;
import com.refactortalentscrm.dto.common.ResultDTO;
import java.util.List;
import java.util.Optional;





public interface NoteService extends GenericService<Note, Integer> {

	List<Note> findAll();

	ResultDTO addNote(NoteDTO noteDTO, RequestDTO requestDTO);

	ResultDTO updateNote(NoteDTO noteDTO, RequestDTO requestDTO);

    Page<Note> getAllNotes(Pageable pageable);

    Page<Note> getAllNotes(Specification<Note> spec, Pageable pageable);

	ResponseEntity<NotePageDTO> getNotes(NoteSearchDTO noteSearchDTO);
	
	List<NoteDTO> convertNotesToNoteDTOs(List<Note> notes, NoteConvertCriteriaDTO convertCriteria);

	NoteDTO getNoteDTOById(Integer noteId);







}





